Networking/Networking/Network/Network.swift

```diff
@@ -18,6 +18,7 @@ public protocol MultipartFormData {
 /// Unit Testing target, and inject mocked up responses.
 ///
 public protocol Network {
+    typealias ResponseHeaders = [String: String]
```

Comment:
Right now it's a dictionary for O(1) lookup, did you mean to convert it to convert the key-value pairs to an array? Or having a struct with a dictionary field?

Reply:
I meant something like
```
struct ResponseHeader {
    let name: String
    let value: String
}

public protocol Network {
    func responseDataAndHeaders(for request: URLRequestConvertible) async throws -> (Data, [ResponseHeader])
}
```

Not a strong requirement, just wondered whether you thought it would make the code clearer. I didn't really consider performance, but the array should be small.

Reply:
Personally, I feel like `[String: String]` is pretty similar to the key-value pairs array both with the need to look up and optional return value, and with guaranteed performance. Maybe we can keep it simple with a dictionary for now, and update to a struct if we have more use cases of the headers.